src/swoop/api/routers/processes.py

```diff
@@ -137,43 +139,44 @@ async def execute_process(
     except KeyError:
         raise HTTPException(status_code=404, detail="Process not found")
 
-    # Generate a UUID
-    action_uuid = uuid.uuid4()
-
-    # Write to object storage
-
-    request.app.state.io.put_object(
-        object_name=f"executions/{action_uuid}/input.json",
-        object_content=json.dumps(payload),
-    )
+    hashed_pl = workflow.hash_payload(payload=payload)
 
     async with request.app.state.readpool.acquire() as conn:
         async with conn.transaction():
-            # Insert a dummy row into payload_cache table
-            # To be removed later
-            pl_uuid = "debeb36c-e09e-41c3-bdc5-596287fe724a"
-
             q, p = render(
-                "INSERT INTO swoop.payload_cache (:values__names) VALUES :values",
-                values=Values(
-                    payload_uuid=pl_uuid,
-                    workflow_version=2,
-                    workflow_name="mirror",
-                ),
+                "SELECT swoop.check_cache(:plhash,\
+                :wf_version::smallint,:wf_name, :invalid::TIMESTAMPTZ)",
+                plhash=hashed_pl,
+                wf_version=workflow.version,
+                wf_name=workflow.name,
+                invalid=datetime.fromisoformat("2023-06-05T15:49:03+00:00"),
             )
-
-            await conn.execute(q, *p)
-
-            q, p = render(
-                "INSERT INTO swoop.action (:values__names) VALUES :values",
-                values=Values(
-                    action_uuid=action_uuid,
-                    action_type="workflow",
-                    action_name=workflow.name,
-                    handler_name=workflow.handler,
-                    payload_uuid=pl_uuid,
-                ),
-            )
-            await conn.execute(q, *p)
-
+            record = await conn.fetchrow(q, *p)
+            rec = record[0]
+
+            if not rec[0]:
+                return RedirectResponse(
+                    request.url_for("get_job_status", job_id=rec[1]),
+                    status_code=303,
+                )
+            else:
+                action_uuid = rec[2]
+                q, p = render(
+                    "INSERT INTO swoop.action (:values__names) VALUES :values",
+                    values=Values(
+                        action_uuid=action_uuid,
+                        action_type="workflow",
+                        action_name=workflow.name,
+                        handler_name=workflow.handler,
+                        payload_uuid=rec[1],
+                    ),
+                )
+                await conn.execute(q, *p)
```

Comment:
What is inserting the payload items and relations?

I kinda expected if you were using a function that it would be inserting anything and everything, else it would insert nothing.

We might need to talk more about what all is/needs to happen here.